modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java

```diff
@@ -244,16 +218,12 @@ protected void registerConfiguration(ConfigurationInstance configuration) {
 
   @Override
   public List<ConfigurationInstance> getExpired() {
-    cacheWriteLock.lock();
-    try {
-      return cache.entrySet().stream().filter(entry -> isExpired(entry.getValue())).map(entry -> {
-        cache.remove(entry.getKey());
-        unRegisterConfiguration(entry.getValue());
-        return entry.getValue();
-      }).collect(toImmutableList());
-    } finally {
-      cacheWriteLock.unlock();
-    }
+    ConcurrentMap<ResolverResultAndEvent, ConfigurationInstance> entries = cache.asMap();
+    return entries.entrySet().stream().filter(entry -> isExpired(entry.getValue())).map(entry -> {
+      cache.invalidate(entry.getKey());
```

Comment:
I think there should be a synchronization between the expiration of entries and the retrieval of entries. So that no config is retrieve right before it is being expired.